controller/cmd/destination/main.go

```diff
@@ -47,25 +47,21 @@ func main() {
 	)
 
 	done := make(chan struct{})
+	ready := make(chan struct{})
 
 	server, lis, err := destination.NewServer(*addr, *k8sDNSZone, *enableTLS, k8sAPI, done)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	go func() {
-		err := k8sAPI.Sync()
-		if err != nil {
-			log.Fatal(err.Error())
-		}
-	}()
+	go k8sAPI.Sync(ready)
 
 	go func() {
 		log.Infof("starting gRPC server on %s", *addr)
 		server.Serve(lis)
 	}()
 
-	go prometheus.NewMetricsServer(*metricsAddr)
+	go admin.StartServer(*metricsAddr, ready)
```

Comment:
something to consider down the road, rename `metricsAddr` to `adminAddr`.

Reply:
Hmm, yeah, for consistency we'd probably also want to rename the `-metrics-addr` flag throughout the code base, but I didn't do that as part of this branch.